kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-api/src/main/java/org/kie/pmml/compiler/api/provider/ModelImplementationProvider.java

```diff
@@ -33,6 +34,8 @@
 
     PMML_MODEL getPMMLModelType();
 
+    Class<? extends KiePMMLModel> getKiePMMLModelClass();
```

Comment:
Let's force this generic to be the same of the one declared above

```suggestion
    Class<E> getKiePMMLModelClass();
```